delatore/sources/awx_api.py

```diff
@@ -87,17 +93,22 @@ async def start(self, stop_event: asyncio.Event):
         await self.client.subscribe(self.params().topic_in)
         await super().start(stop_event)
 
-    async def get_templates(self, filters: dict = None) -> Optional[list]:
+    async def get_templates(self, filters: dict = None, depth: int = 1) -> Optional[list]:
         """Returns list of all job templates for csm organization
 
         This methods support Ansible tower filtering
         https://docs.ansible.com/ansible-tower/latest/html/towerapi/filtering.html
         """
         async with self.get_session() as session:
-            async with session.get(self.params().host + '/job_templates', params=filters, timeout=5) as response:
+            async with session.get(self.params().host + '/job_templates', params=filters, timeout=15) as response:
                 response_data = await response.json()
             assert response.status == 200, f'Expected response 200, got {response.status}'
         try:
+            if depth > 1:
+                container = []
+                for i in range(depth):
+                    container.append(create_record(response_data['results'][0], i))
```

Comment:
`response_data['results'][0]` is retrieved every iteration, should be moved out of cycle